src/Storage/Commands.Storage/Common/StorageCloudCmdletBase.cs

```diff
@@ -622,5 +622,32 @@ protected override void StopProcessing()
             cancellationTokenSource.Cancel();
             base.StopProcessing();
         }
+
+        /// <summary>
+        /// true if FIPS policy is enabled on the current machine
+        /// </summary>
+        protected static bool FIPSEnabled
+        {
+            get
+            {
+                if (fIPSEnabled.HasValue)
+                {
+                    return fIPSEnabled.Value;
+                }
+
+                try
+                {
+                    System.Security.Cryptography.MD5.Create();
+                }
+                catch (System.Reflection.TargetInvocationException)
+                {
+                    fIPSEnabled = true;
+                    return true;
+                }
+                fIPSEnabled = false;
```

Comment:
vertical space

Reply:
```c#
public static bool FipsEnabled {get;} = IsFIPSEnabled();

internal static bool ISFIPSEnabled()
{
                try
                {
                    System.Security.Cryptography.MD5.Create();
                    return false;
                }
              catch (System.Reflection.TargetInvocationException)
                {
                    return true;
                }
}
```

Reply:
I think we can take advantage of the fact that the static initializer will run once to simploify the code - so we cans et the value and then use it later

Reply:
Fixed as the code from you. Thanks!